### PR TITLE
Set "Auto" as default input size mode

### PR DESCRIPTION
**Pull request by goodsong81**
### Summary

* Refine auto input size adaptation
* Set "Auto" as default mode
* Fix tests
* Update documentation

### How to test

* Integration & e2e tests

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [x] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [x] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [x] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [x] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by sungmanc — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -60,13 +61,13 @@ OpenVINO™ Training Extensions supports the [following learning methods](https:
 - **Semi-supervised learning**
 - **Self-supervised learning**
 
-OpenVINO™ Training Extensions will provide the following features in coming releases:
+OpenVINO™ Training Extensions provides the following usability features:
 
+- [Auto-configuration](https://openvinotoolkit.github.io/training_extensions/latest/guide/explanation/additional_features/auto_configuration.html). OpenVINO™ Training Extensions analyzes provided dataset and selects the proper task and model with appropriate input size to provide the best accuracy/speed trade-off. It will also make a random auto-split of your dataset if there is no validation set provided.
+- [Datumaro](https://openvinotoolkit.github.io/datumaro/stable/index.html) data frontend: OpenVINO™ Training Extensions supports the most common academic field dataset formats for each task. We constantly working to extend supported formats to give more freedom of datasets format choice.
```
Comment (eunwoosh) [inline README.md]:
```suggestion
- [Datumaro](https://openvinotoolkit.github.io/datumaro/stable/index.html) data frontend: OpenVINO™ Training Extensions supports the most common academic field dataset formats for each task. We are constantly working to extend supported formats to give more freedom of datasets format choice.
```
Maybe it's a typo?

File: src/otx/algorithms/detection/configs/rotated_detection/configuration.yaml [hunk 2]
```diff
@@ -350,28 +350,6 @@ algo_backend:
       type: UI_RULES
     visible_in_ui: false
     warning: null
-  storage_cache_scheme:
```
Comment (eunwoosh) [inline src/otx/algorithms/detection/configs/rotated_detection/configuration.yaml]: It's just a question. Why is it removed?

**Review by eunwoosh — COMMENTED** [review]
Thanks for your work! I left some comments. Please take a look.

File: src/otx/algorithms/detection/configs/rotated_detection/configuration.yaml (hunk 2 above)
Comment (goodsong81) [reply src/otx/algorithms/detection/configs/rotated_detection/configuration.yaml]: Thought it's irrelevant, but I've reverted. Thanks.

File: README.md (hunk 1 above)
Comment (goodsong81) [reply README.md]: Fixed, thanks.

**Review by eunwoosh — APPROVED** [review]

**Review by sungmanc — APPROVED** [review]

**Pull request merged**